A new rule for automatic reconciliation and a cron [rebased]

lepistone opened the pull request:
A rebased version of https://github.com/OCA/bank-statement-reconcile/pull/7

Discussion continues from there.

mdietrichc2c committed:
[ADD] new reconciliation rule using bank statements

mdietrichc2c committed:
[FIX] updated copyrights

mdietrichc2c committed:
[FIX] optimize new rule

nbessi committed:
[MRG] [ADD] a cron to run reconciles.
  By default it runs one reconcile after the other to be more atomic
  and memory friendly. But you can set run_all agrument to have all reconcile
  run at once

nbessi committed:
Add missing import

Due to merge trouble

guewen commented on account_advanced_reconcile_bank_statement/advanced_reconciliation.py:
```diff
@@ -0,0 +1,82 @@
+        result += ["account_bank_statement.name as statement_name"]
+        return result
+
+    def _from(self, rec, *args, **kwargs):
```
It should extend and not override (I'll make the change).

Commit:
Extend methods instead of overriding them entirely.

Allows to stay compatible if the original methods change, for instance if new fields are added.

guewen commented on account_advanced_reconcile_bank_statement/advanced_reconciliation.py:
```diff
@@ -0,0 +1,79 @@
+        credit_lines = self._query_credit(cr, uid, rec, context=context)
+        debit_lines = self._query_debit(cr, uid, rec, context=context)
+        return self._rec_auto_lines_advanced(
+                cr, uid, rec, credit_lines, credit_lines + debit_lines,
```
Shouldn't it be

```
credit_lines + debit_lines,
credit_lines + debit_lines,
```

according to https://github.com/guewen/bank-statement-reconcile/commit/279a176e22bcca4db214dae665abafb02cfd672d

nbessi commented:
:+1:

bwrsandman commented:
Needs to be rebased

Commit:
add debit lines for matching

according to
https://github.com/guewen/bank-statement-reconcile/commit/279a176e22bcca4db214dae665abafb02cfd672d

guewen commented:
Rebased

Commit:
pep8 fixes

Commit:
Cron fails to run when there is no "last history".

Fixes #39

lepistone commented:
rebased again

lepistone committed:
removed outdated comment

lepistone committed:
fix author in header

I asked Romain: he is not the author, the header was copy-pasted by
mistake.

lepistone commented:
A committed a few trivial fixes. The original author was @mdietrichc2c, and I vote :+1: !

@guewen you fixed something here, too. Do you approve as it stands now?

Thanks!

guewen commented:
Aye :+1:

lepistone commented:
Thanks. We have 3 approvals other than the original author, so I merge.
